tests/image_processing_tests/CMakeLists.txt

```diff
@@ -21,34 +21,31 @@
 SET(PROJ_NAME kpsr_mem_${PROJ_MAIN_NAME}_tests)
 PROJECT(${PROJ_NAME})
 
-# Include directories with headers
-#---------------------------------------------------#
-INCLUDE_DIRECTORIES( ${${PROJ_MAIN_NAME}_PATH_MAIN}/modules/common )
-INCLUDE_DIRECTORIES( ${${PROJ_MAIN_NAME}_PATH_MAIN}/modules/${PROJ_MAIN_NAME}/include )
-INCLUDE_DIRECTORIES( ${${PROJ_MAIN_NAME}_PATH_MAIN}/modules/mem_mdlw/include )
-INCLUDE_DIRECTORIES( ${CMAKE_CURRENT_SOURCE_DIR}/include )
-INCLUDE_DIRECTORIES( ${KLEPSYDRA_INCLUDE_DIRS} )
-
 # Source
 #---------------------------------------------------#
 FILE(GLOB ${PROJ_NAME}_SRC
     "src/*.cpp"
 )
 
-# Link libraries with Project
-#---------------------------------------------------#
-LINK_DIRECTORIES()
-
 # Create Library
 #---------------------------------------------------#
 ADD_EXECUTABLE(${PROJ_NAME} ${${PROJ_NAME}_HEADERS} ${${PROJ_NAME}_SRC} ) 
 
 TARGET_LINK_LIBRARIES(${PROJ_NAME}
-  ${KLEPSYDRA_CORE_LIBRARIES}
+  PUBLIC
   ${KPSR_ROBOTICS_CORE_LIBRARIES}
   kpsr_${PROJ_MAIN_NAME}
   gtest_main ${OpenCV_LIBS}
-  atomic
 )
 
-add_test(NAME ${PROJ_NAME} COMMAND ${PROJ_NAME})
+# Include directories with headers
+#---------------------------------------------------#
+TARGET_INCLUDE_DIRECTORIES(${PROJ_NAME}
+  PRIVATE
+  ${${PROJ_MAIN_NAME}_PATH_MAIN}/modules/common
+  ${${PROJ_MAIN_NAME}_PATH_MAIN}/modules/${PROJ_MAIN_NAME}/include
```

Comment:
this line will not be necessary if the include directories of `kpsr_${PROJ_MAIN_NAME}` are public.